.github/workflows/update-generated-files.yaml

```diff
@@ -0,0 +1,66 @@
+name: Update generated files
+
+on:
+  push:
+    branches:
+      - develop
+
+env:
+  BRANCH_NAME: sync-generated-files
+
+jobs:
+  update_genereated_files:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Build
+        run: npm run build
+
+      - name: Check for changes
+        id: changes
+        run: |
+          changes=$(git status --porcelain)
+          # see https://unix.stackexchange.com/a/509498
+          echo $changes | grep . && echo "Changes detected" || echo "No changes"
+          echo ::set-output name=changes::"$changes"
+
+      - name: Check branch exists
+        id: branchExists
+        if: steps.changes.outputs.changes
+        run: |
+          exists=$(git ls-remote --heads origin $BRANCH_NAME)
+          echo $exists | grep . && echo "Branch '$BRANCH_NAME' already exists on remote" || echo "Branch does not exists in remote"
+          echo ::set-output name=exists::"$exists"
+
+      - name: Create pull request
+        if: ${{ steps.changes.outputs.changes && !steps.branchExists.outputs.exists }}
```

Comment:
You can see a working example of using the the step outputs:

- [no changes](https://github.com/straker/issue-template-tests/runs/8076907049?check_suite_focus=true)
- [changes and branch does not exist](https://github.com/straker/issue-template-tests/runs/8077228637?check_suite_focus=true)
- [changes and branch already exists](https://github.com/straker/issue-template-tests/runs/8077261494?check_suite_focus=true)